packages/web-components/src/checkbox/checkbox.ts

```diff
@@ -1,118 +1,466 @@
- * @slot - The default slot for the label
- * @csspart control - The element representing the visual checkbox control
- * @csspart label - The label
  * @fires change - Emits a custom change event when the checked state changes
+ * @fires input - Emits a custom input event when the checked state changes
  *
  * @public
  */
-export class Checkbox extends FormAssociatedCheckbox {
+export class Checkbox extends FASTElement {
+  /**
+   * Indicates that the element should get focus after the page finishes loading.
+   * @see The {@link https://developer.mozilla.org/docs/Web/HTML/Element/input#autofocus | `autofocus`} attribute
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `autofocus`
+   */
+  @attr({ mode: 'boolean' })
+  public autofocus!: boolean;
+
+  /**
+   * The element's current checked state.
+   *
+   * @public
+   */
+  public get checked(): boolean {
+    Observable.track(this, 'checked');
+    return this._checked;
+  }
+
+  public set checked(state: boolean) {
+    this._checked = state;
+
+    this.setFormValue(state ? this.value : null);
+    this.setValidity();
+    this.setAriaChecked();
+    toggleState(this.elementInternals, 'checked', state);
+
+    Observable.notify(this, 'checked');
+  }
+
+  /**
+   * The default slotted nodes.
+   *
+   * @internal
+   */
+  @observable
+  public defaultSlottedNodes: Node[] = [];
+
+  /**
+   * The element's disabled state.
+   * @public
+   * @remarks
+   * HTML Attribute: `disabled`
+   */
+  @attr({ mode: 'boolean' })
+  public disabled: boolean = false;
+
+  /**
+   * The id of a form to associate the element to.
+   * @see The {@link https://developer.mozilla.org/docs/Web/HTML/Element/input#form | `form`} attribute
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `form`
+   */
+  @attr({ attribute: 'form' })
+  public formAttribute?: string;
+
+  /**
+   * The `indeterminate` state of the control.
+   *
+   * @public
+   */
+  @observable
+  public indeterminate?: boolean;
+
+  /**
+   * Synchronizes the element's `indeterminate` state with the internal ElementInternals state.
+   *
+   * @internal
+   */
+  public indeterminateChanged(prev: boolean, next: boolean): void {
+    this.setAriaChecked();
+    toggleState(this.elementInternals, 'indeterminate', next);
+  }
+
+  /**
+   * The element's checked state.
+   * @public
+   * @remarks
+   * HTML Attribute: `checked`
+   */
+  @attr({ attribute: 'checked', mode: 'boolean' })
+  public initialChecked?: boolean;
+
+  /**
+   * Updates the form value when the checked state changes.
+   *
+   * @internal
+   */
+  public initialCheckedChanged(prev: boolean | undefined, next: boolean): void {
+    if (!this.dirtyChecked) {
+      this.checked = next;
+    }
+  }
+
+  /**
+   * The initial value of the input.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `value`
+   */
+  @attr({ attribute: 'value', mode: 'fromView' })
+  public initialValue: string = 'on';
```

Comment:
`initialChecked` is optional (false) but `initialValue` is `'on'`... should those match?

Reply:
The `initialValue` property tracks the `value` attribute, which contains the data that's sent when the form is submitted. The value is `"on"` by default.

The `initialChecked` property tracks the `checked` attribute, which is only used to set the initial checkedness (via the markup).

When the form is submitted, the checkbox's name-value pair is sent with the form data if the checked state is true. In a GET request, this would appear as `name=value`. If the checked state is false, then nothing gets sent.

If this element were in a form, its data would be sent as `favorite_fruit=banana`:
```html
<fluent-checkbox checked name="favorite_fruit" value="banana"></fluent-checkbox>
```


Some additional context for the naming: while changing the field value of the checkbox isn't really doable by the user, it can still be changed programmatically by setting the `value` property. Separating the `value` attribute and the `value` property (binding to `initialValue`) allows for resetting the value to its initial state if a form reset is performed. The same goes for the `checked` state - if we're only tracking the presence of the _attribute_, we have no way to reliably reset the checkedness to match what it was when originally created.

Reply:
Awesome! Thanks for the clarification. I forgot you can set custom values on checkboxes.